src/equations/laplace_diffusion_2d.jl

```diff
@@ -0,0 +1,53 @@
+@doc raw"""
+    LaplaceDiffusion2D(diffusivity, equations)
+
+`LaplaceDiffusion2D` represents a scalar diffusion term ``\nabla \cdot (\kappa\nabla u))``
+with diffusivity ``\kappa`` applied to each solution component defined by `equations`.
+
+If `equations` is not specified, a scalar `LaplaceDiffusion2D` is returned.
+"""
+struct LaplaceDiffusion2D{E, N, T} <: AbstractLaplaceDiffusionEquations{2, N}
+  diffusivity::T
+  equations::E
+end
+
+function LaplaceDiffusion2D(diffusivity, equations=nothing)
```

Comment:
That's a good point. Changing to `LaplaceDiffusion2D(diffusivity, equations::AbstractEquations)`...

Reply:
I don't think you need the dispatch on `AbstractEquations` - I think it would be enough if you lose the `if equations isa AbstractEquations` part and just assume it is one.